lopper/assists/baremetallinker_xlnx.py

```diff
@@ -177,8 +178,22 @@ def xlnx_generate_bm_linker(tgt_node, sdt, options):
     except IndexError:
         pass
 
-    src_dir = options['args'][1].rstrip(os.path.sep)
-    appname = os.path.basename(os.path.dirname(src_dir))
+    src_dir = options['args'][1]
+    app_path = utils.get_dir_path(src_dir.rstrip(os.sep))
+    appname = utils.get_base_name(app_path)
+    yaml_file = os.path.join(app_path, "data", f"{appname}.yaml")
+    stack_size = None
+    heap_size = None
+    if not utils.is_file(yaml_file):
+        print(f"{appname} doesn't have yaml file")
+    else:
+        schema = utils.load_yaml(yaml_file)
+        if schema.get("linker_constraints"):
+            if schema.get("linker_constraints")["stack"]:
```

Comment:
Hi @kedareswararao ,
I guess you intend to use:
    schema["linker_constraints"].get("stack")

And same for heap.

Reply:
Thanks fixed it now...